packages/io/src/chain-source/nervos.ts

```diff
@@ -20,25 +23,43 @@ export class NervosChainSource implements ChainSource {
     return header.number
   }
 
-  getAllLiveCellsWithWitness = async (
-    lockScript: Script,
-    typeScript?: Script | undefined,
-    startBlock?: BI,
-    endBlock?: BI,
-  ) => {
+  getAllLiveCellsWithWitness = async (lockScript?: Script, typeScript?: Script, startBlock?: BI, endBlock?: BI) => {
```

Comment:
Two objects with named fields would be clearer, otherwise `undefined` should be passed in explicitly.

```ts
getAllLiveCellsWithWitnesses = async (scripts: Partial<Record<'lock' | 'type', Script>>, blockRange?: Partial<Record<'start' | 'end', BI>> => {}
```